src/Compilers/CSharp/Test/Symbol/Symbols/Retargeting/RetargetingTests.cs

```diff
@@ -750,6 +751,262 @@ class C1<T>
+
+            assert(param2Consistent.typeConsistent,
+                   param2Original.Type,
+                   param2Retargeted.Type);
+            assert(param2Consistent.refModConsistent,
+                   getModifierTypeSymbol(param2Original.RefCustomModifiers),
+                   getModifierTypeSymbol(param2Retargeted.RefCustomModifiers));
+            assert(param2Consistent.typeModConsistent,
+                   getModifierTypeSymbol(param2Original.TypeWithAnnotations.CustomModifiers),
+                   getModifierTypeSymbol(param2Retargeted.TypeWithAnnotations.CustomModifiers));
+
+            static MethodSymbol getMethodSymbol(CSharpCompilation compilation)
+            {
+                var c = compilation.GetTypeByMetadataName("Source");
+                return c.GetMethod("M");
+            }
+
+            static TypeSymbol getModifierTypeSymbol(ImmutableArray<CustomModifier> modifiers)
+                => ((CSharpCustomModifier)modifiers.Single()).ModifierSymbol;
+
+            void assert(bool consistent, TypeSymbol originalType, TypeSymbol retargetedType)
+            {
+                Assert.False(originalType.IsErrorType());
+                Assert.False(retargetedType.IsErrorType());
+                if (consistent)
+                {
+                    Assert.Same(consistentAssembly, originalType.ContainingAssembly);
+                    Assert.Same(consistentAssembly, retargetedType.ContainingAssembly);
+                }
+                else
+                {
+                    Assert.Same(retargetedAssembly1, originalType.ContainingAssembly);
+                    Assert.Same(retargetedAssembly2, retargetedType.ContainingAssembly);
+                }
+            }
+
+            static (AssemblySymbol retargetedAssembly1, AssemblySymbol retargetedAssembly2, AssemblySymbol consistentAssembly, CSharpCompilation originalComp, CSharpCompilation retargetedComp)
+                getFunctionPointerRetargetingDefinitions(string mIlSignature, string mOverriddenSignature)
+            {
+                var retargetedSource = @"public class R {{}}";
+                var retargetedIdentity = new AssemblyIdentity("Ret", new Version(1, 0, 0, 0), isRetargetable: true);
+                var standardReference = TargetFrameworkUtil.StandardReferences.ToArray();
+                var retargeted1 = CreateCompilation(retargetedIdentity, new[] { retargetedSource }, references: standardReference);
+                var retargeted1Ref = retargeted1.ToMetadataReference();
+                var retargeted2 = CreateCompilation(retargetedIdentity.WithVersion(new Version(2, 0, 0, 0)), new[] { retargetedSource }, references: standardReference);
+                var retargeted2Ref = retargeted2.ToMetadataReference();
+
+                var consistent = CreateCompilation("public class C {}", assemblyName: "Con", targetFramework: TargetFramework.Standard);
```

Comment:
Because I had to add an explicit StandardReference above (because that CreateCompilation overload does not add default references), I'm being explicit in the others rather than relying on any change to CreateCompilation to change for all of the calls.